Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelDetectorObjectFactory.hpp

```diff
@@ -0,0 +1,78 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelDetectorElement.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelToDetectorVolume.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelTree.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <GeoModelHelpers/getChildNodesWithTrf.h>
+
+#include "GeoModelKernel/GeoDefinitions.h"
+
+class GeoShape;
+struct GeoModelTree;
+class Surface;
+namespace Acts {
+class GeoModelDetectorObjectFactory {
+ public:
+  using GeoModelBoundingBox = std::shared_ptr<Experimental::DetectorVolume>;
+
+  struct Options {
+    std::vector<std::string> queries = {};
+  };
+  struct Config {
+    // /// List for names to match
+    std::vector<std::string> nameList;
+    /// List for materials to match
+    std::vector<std::string> materialList;
+
+    /// boolean flag to build subvolumes
+    bool convertSubVolumes = false;
+
+    /// flag to build the desired bounding Boxes
+    std::string convertBox;
+  };
+  struct Cache {
+    /// The created detector elements and their surfaces
+    std::vector<GeoModelSensitiveSurface> sensitiveSurfaces;
+    /// The created representation of bounding box
+    std::vector<GeoModelBoundingBox> boundingBoxes;
+  };
+
+  GeoModelDetectorObjectFactory(
+      const Config& cfg,
+      std::unique_ptr<const Logger> mlogger = getDefaultLogger(
+          "GeoModelDetectorObjectFactory", Acts::Logging::WARNING));
+
+  void construct(Cache& cache, const GeometryContext& gctx,
+                 const GeoModelTree& geoModelTree, const Options& options);
+
+  void convertSensitive(PVConstLink geoPV, const Acts::Transform3& transform,
+                        std::vector<GeoModelSensitiveSurface>& sensitives);
+
+  std::vector<GeoChildNodeWithTrf> findAllSubVolumes(PVConstLink geoPV);
+
+  bool convertBox(std::string name);
+  bool matches(const std::string& name, PVConstLink physvol);
```

Comment:
can we have this function as lamda as it was before?

Reply:
I'm not sure since we call from both the `construct` and the `findAllSubVolumes` member functions. I could pass it as an argument to the `convertFpv` to have it then passed down to the `findAllSubVolumes`but that seems unnecessarily complicated to me. Or is there a  benefit in defining it as it was in the surfaceFactory?